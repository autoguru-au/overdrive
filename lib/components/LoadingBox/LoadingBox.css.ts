import { keyframes, style } from '@vanilla-extract/css';

import { themeContractVars as vars } from '../../themes/theme.css';

export const root = style({
	minHeight: vars.typography.size['3'].fontSize,
});

const blinker = keyframes({
	'0%': {
		opacity: 0.4,
	},

	'50%': {
		opacity: 1,
	},

	'100%': {
		opacity: 0.4,
	},
});

export const blinking = style({
	animationDuration: '1.5s',
	animationTimingFunction: vars.animation.easing.standard,
	animationIterationCount: 'infinite',
	animationName: blinker,
});
