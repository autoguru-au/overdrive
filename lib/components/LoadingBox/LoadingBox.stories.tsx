import { Meta, StoryObj } from '@storybook/react';
import React from 'react';

import { boxArgTypes } from '../../stories/shared/argTypes-box';

import { LoadingBox } from '.';

export default {
	title: 'Components/Loading Box',
	component: LoadingBox,
	decorators: [
		(story) => (
			<div style={{ width: '50%', height: '100%', margin: '0 auto' }}>
				{story()}
			</div>
		),
	],
	parameters: { chromatic: { disable: true } },
	argTypes: boxArgTypes,
} satisfies Meta<typeof LoadingBox>;

type Story = StoryObj<typeof LoadingBox>;

export const Standard: Story = {
	args: {},
};

export const BlinkingOff: Story = {
	args: {
		blinking: false,
	},
};

export const RandomWidth: Story = {
	args: {
		randomWidth: true,
	},
};
