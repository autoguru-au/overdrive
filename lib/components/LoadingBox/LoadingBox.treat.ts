import { style } from 'treat';

export const root = style(theme => ({
	width: '100%',
	display: 'inline-block',
	minHeight: theme.typography.size['3'].fontSize,
}));

export const blinking = style(theme => ({
	animationDuration: '1.5s',
	animationTimingFunction: theme.animation.easing.standard,
	animationIterationCount: 'infinite',

	'@keyframes': {
		'0%': {
			opacity: 0.4,
		},

		'50%': {
			opacity: 1,
		},

		'100%': {
			opacity: 0.4,
		},
	},
}));
