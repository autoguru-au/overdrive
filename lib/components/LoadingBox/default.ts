export { LoadingBox as default } from './LoadingBox';
