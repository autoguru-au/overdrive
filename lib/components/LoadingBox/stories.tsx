import { ComponentMeta, ComponentStory } from '@storybook/react';
import * as React from 'react';
import { ComponentProps } from 'react';

import { boxArgTypes } from '../Box/argTypes';

import { LoadingBox } from '.';

export default {
	title: 'Components/Loading/Box',
	component: LoadingBox,
	decorators: [
		(story) => (
			<div style={{ width: '50%', height: '100%', margin: '0 auto' }}>
				{story()}
			</div>
		),
	],
	parameters: { chromatic: { disable: true } },
	argTypes: boxArgTypes,
} as ComponentMeta<typeof LoadingBox>;

const Template: ComponentStory<typeof LoadingBox> = (args) => (
	<LoadingBox {...args} />
);

const standardProps: ComponentProps<typeof LoadingBox> = {};
export const standard = Template.bind(standardProps);
standard.args = standardProps;

const blinkingOffProps: ComponentProps<typeof LoadingBox> = {
	blinking: false,
};
export const blinkingOff = Template.bind(blinkingOffProps);
blinkingOff.args = blinkingOffProps;

const randomWidthProps: ComponentProps<typeof LoadingBox> = {
	randomWidth: true,
};
export const randomWidth = Template.bind(randomWidthProps);
randomWidth.args = randomWidthProps;
