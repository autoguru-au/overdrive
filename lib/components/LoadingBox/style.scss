@import '../../theme/tokens/facade';

:root {
	--global--min--loading--box--height: 14px;
	--loading--box--fade-level: 0.5;
}

@keyframes fade {
	0% {
		opacity: var(--loading--box--fade-level);
	}

	50% {
		opacity: 1;
	}

	100% {
		opacity: var(--loading--box--fade-level);
	}
}

.root {
	width: 100%;
	min-height: var(--global--min--loading--box--height);
	background-color: getColour(gray, 200);
}

.blinking {
	animation: fade 1.5s ease-in-out infinite;
}
