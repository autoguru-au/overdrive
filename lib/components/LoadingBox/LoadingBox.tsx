import * as React from 'react';
import { ComponentProps, FunctionComponent } from 'react';

import { Box } from '../Box';

import * as styles from './LoadingBox.css';

export interface Props extends Omit<ComponentProps<typeof Box>, 'width'> {
	className?: string;
	randomWidth?: boolean;
	blinking?: boolean;
}

export const LoadingBox: FunctionComponent<Props> = ({
	className = '',
	randomWidth = false,
	blinking = true,
	backgroundColour = 'gray200',
	display = 'block',
	is = 'span',
	...boxStyles
}) => (
	<Box
		is={is}
		display={display}
		width={randomWidth ? undefined : 'full'}
		backgroundColour={backgroundColour}
		{...boxStyles}
		className={[styles.root, blinking && styles.blinking, className]}
		style={{
			width: randomWidth ? getRandomIntWidth(60, 40) : undefined,
		}}
	/>
);

const getRandomIntWidth = (max: number, min: number) =>
	`${Math.random() * (max - min) + min}%`;
