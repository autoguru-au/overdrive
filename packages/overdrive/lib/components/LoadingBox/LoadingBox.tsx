import * as React from 'react';
import { FunctionComponent } from 'react';
import { useStyles } from 'react-treat';

import type { BoxStyleProps } from '../Box';
import { Box } from '../Box';
import * as styleRefs from './LoadingBox.treat';

export interface Props
	extends Partial<Pick<BoxStyleProps, 'height' | 'display'>> {
	className?: string;
	randomWidth?: boolean;
	blinking?: boolean;
}

export const LoadingBox: FunctionComponent<Props> = ({
	className = '',
	randomWidth = false,
	blinking = true,
	display = 'block',
	height,
}) => {
	const styles = useStyles(styleRefs);

	return (
		<Box
			is="span"
			width={randomWidth ? undefined : 'full'}
			backgroundColour="gray200"
			display={display}
			height={height}
			className={[styles.root, blinking && styles.blinking, className]}
			style={{
				width: randomWidth ? getRandomIntWidth(60, 40) : undefined,
			}}
		/>
	);
};

const getRandomIntWidth = (max: number, min: number) =>
	`${Math.random() * (max - min) + min}%`;
