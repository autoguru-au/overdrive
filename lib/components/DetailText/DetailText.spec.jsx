import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { DetailText, DetailTextLoader, EDetailTextSize } from './';
import { LoadingBox } from '../LoadingBox';

describe('<DetailTextLoader />', () => {
	it('should not throw', () =>
		expect(() => shallow(<DetailTextLoader />)).not.toThrow());

	it('should all a loding box component', () => {
		const detailTextLoader = mount(<DetailTextLoader />);
		expect(detailTextLoader.find(LoadingBox).exists()).toEqual(true);
		detailTextLoader.unmount();
	});

	it('should match snapshot', () => {
		expect(render(<DetailTextLoader />)).toMatchSnapshot();
	});
});

describe('<DetailText />', () => {
	it('should not throw', () =>
		expect(() => shallow(<DetailText />)).not.toThrow());

	it('should match snapshot without label', () => {
		expect(shallow(<DetailText />)).toMatchSnapshot();
	});

	it('should match snapshot with label for size 1', () => {
		expect(
			render(
				<DetailText size={EDetailTextSize.Detail1}>
					Hello World
				</DetailText>
			)
		).toMatchSnapshot();
	});

	it('should match snapshot with label for size 2', () => {
		expect(
			render(
				<DetailText size={EDetailTextSize.Detail2}>
					Hello World
				</DetailText>
			)
		).toMatchSnapshot();
	});

	it('should match snapshot with label for size 3', () => {
		expect(
			render(
				<DetailText size={EDetailTextSize.Detail3}>
					Hello World
				</DetailText>
			)
		).toMatchSnapshot();
	});

	it('should pass on className to dom element', () => {
		const detailText = mount(
			<DetailText className="detailText-class" label="Hello World" />
		);
		expect(
			detailText.find('span').hasClass('detailText-class')
		).toBeTruthy();
		detailText.unmount();
	});

	it('should use a span dom element ', () => {
		const detailText = shallow(
			<DetailText className="detailText-class">Hello World</DetailText>
		);
		expect(detailText.type()).toEqual('span');
	});

	it('should display the child text', () => {
		const detailText = mount(
			<DetailText className="detailText-class">Hello World</DetailText>
		);
		expect(detailText.text()).toEqual('Hello World');
		detailText.unmount();
	});

	it('should add a detail1 class for detail 1 size', () => {
		const detailText = mount(
			<DetailText size={EDetailTextSize.Detail1}>Hello World</DetailText>
		);
		expect(detailText.find('span').hasClass('detail1')).toBeTruthy();
		detailText.unmount();
	});

	it('should add a detail2 class for detail 2 size', () => {
		const detailText = mount(
			<DetailText size={EDetailTextSize.Detail2}>Hello World</DetailText>
		);
		expect(detailText.find('span').hasClass('detail2')).toBeTruthy();
		detailText.unmount();
	});

	it('should add a detail3 class for detail 3 size', () => {
		const detailText = mount(
			<DetailText size={EDetailTextSize.Detail3}>Hello World</DetailText>
		);
		expect(detailText.find('span').hasClass('detail3')).toBeTruthy();
		detailText.unmount();
	});

	describe('when muted', () => {
		it('should match snapshot', () => {
			expect(
				render(<DetailText muted>Hello World</DetailText>)
			).toMatchSnapshot();
		});
	});
});
