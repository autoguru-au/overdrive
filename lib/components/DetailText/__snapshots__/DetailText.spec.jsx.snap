// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailText /> should match snapshot with label for size 1 1`] = `
<span
  class="root detail1"
>
  Hello World
</span>
`;

exports[`<DetailText /> should match snapshot with label for size 2 1`] = `
<span
  class="root detail2"
>
  Hello World
</span>
`;

exports[`<DetailText /> should match snapshot with label for size 3 1`] = `
<span
  class="root detail3"
>
  Hello World
</span>
`;

exports[`<DetailText /> should match snapshot without label 1`] = `
<span
  className="root detail1"
/>
`;

exports[`<DetailText /> when muted should match snapshot 1`] = `
<span
  class="root detail1 muted"
>
  Hello World
</span>
`;

exports[`<DetailTextLoader /> should match snapshot 1`] = `
<span
  class="root root loadingDetail1 blinking"
/>
`;
